lib/local_daily_digest.js

```diff
@@ -0,0 +1,167 @@
+			});
+			console.log("the countries this profile is either a manger or editor of are: ", requiredLocationsId);
+			return cb();
+		},
+		function (cb){
+			Contact.find({'type':'local'}, function(err, contact){
+				if (err) {
+		          res.send(500, new Error("err"));
+		          return cb(true);
+		        }
+		        if (!contact) {
+		          res.send(404, new Error('No Local Profiles found  not found'));
+		          return cb(true);
+		        }
+			contact.forEach(function(profile){
+				allLocalProfiles.push(profile);
+			});
+	  		return cb();
+		  });
+		},
+		function (cb){
+			allLocalProfilesWDigest.forEach(function(managerProfile){
+				allLocalProfiles.forEach(function(profile){
+					if(managerProfile.locationId == profile.locationId){
+						requiredUsers.push(profile);
+					};
+				});
+			});
+			console.log("the no of users: ", requiredUsers.length);
+			return cb();
+		},
+		function (cb){
+			requiredUsers.forEach(function(user){
+				if(!user){
+					console.log("No users found1");
+					return cb();
+				}
+		    	else 
+	    		{
+	    			 	var currentDate= new Date().getTime();
+						var userCheckInDate = new Date(user.created).getTime();
+						var userUpdateDate = new Date(user.revised).getTime();
+						var updatedTimeDiff = Math.abs((currentDate - userUpdateDate)/86400000);
+					    var checkInDiff = Math.abs((currentDate - userCheckInDate) / 86400000);    
+						if(checkInDiff <= 1){  
+							if(user.status){
+								if(checkedInUsers.length == 0 || checkedInUsers == null)
+									checkedInUsers.push(user);
+								else{
+									for(var i = 0; i < checkedInUsers.length; i++){
+										if(user._id != checkedInUsers[i]._id){
+											if(user.locationId != checkedInUsers[i].locationId){
+												checkedInUsers.push(user);
+											}
+										}
+									}
+								}
+							}
+						}
+						
+						if(updatedTimeDiff <=1 && !user.status){
+							if(checkedOutUsers.length == 0 || checkedOutUsers == null){	
+								checkedOutUsers.push(user);
+							}
+							else{
+								for(var i = 0; i < checkedOutUsers.length; i++){
+									if(user._id != checkedOutUsers[i]._id && user.locationId != checkedOutUsers[i].locationId){
+										checkedOutUsers.push(user);
+									}
+								}
+								
+							}
+						}
+	    		}
+			});
+			return cb();
+		},
+		function (cb){
+			allLocalProfilesWDigest.forEach(function(managerProfile){
+				var checkInTemp = [];
+				var checkOutTemp = [];
+				
+				checkedInUsers.forEach(function(checkedInProfiles){
+					if(managerProfile.locationId == checkedInProfiles.locationId){
+						checkInTemp.push(checkedInProfiles._id);
+					}
+				});
+
+				checkedOutUsers.forEach(function(checkedOutProfiles){
+					if(managerProfile.locationId == checkedOutUsers.locationId){
+						checkOutTemp.push(checkedOutUsers._id);
+					}
+				});
+
+				//Using these ID's we can get the information needed and send an email
+				finalOutput.push({
```

Comment:
The email should be sent out directly in the cron script, not returned in a response.